File: ps-backend/src/main/java/org/commonground/ps/backendapi/model/Note.java
package org.commonground.ps.backendapi.model;

import java.util.Date;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;

import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class Note {

    private Long id;
    @NotNull(message = "Waarde is verplicht")
    @Size(min = 1, max = 5000, message = "Waarde is minimaal 1 en maximaal 5000 tekens")
    private String content;
    private NoteType type;
    private User user;
    private Date dateCreated;
    private boolean visible;
}

File: ps-frontend/src/model/note.ts
import { User } from "./user";

export class NoteType {
  public id?: number;
  public name?: string;
}

export class Note {
  public id?: string;
  public content?: string;
  public type?: NoteType;
  public user?: User;
  public public?: boolean = false;
  public dateCreated?: Date;

  public constructor(content: string = '') {
    this.content = content;
  }

}
